src/variety/bosnianroad.js

```diff
@@ -234,13 +234,47 @@
 	AnsCheck: {
 		checklist: [
 			"checkShadeCellExist+",
+			"check2x2ShadeSection",
 			"checkShadeBranch",
 			"checkShadeDeadEnd",
 			"checkConnectShade",
 			"checkShadeDiagonal",
 			"checkShadeCount+"
 		],
 
+		check2x2ShadeSection: function() {
+			var isShade = function(cell) {
+				return cell.isShade();
+			};
+			var bd = this.board;
+			for (var c = 0; c < bd.cell.length; c++) {
+				var cell = bd.cell[c];
+				if (cell.bx >= bd.maxbx - 1 || cell.by >= bd.maxby - 1) {
+					continue;
+				}
+
+				var bx = cell.bx,
+					by = cell.by;
+				var clist = bd.cellinside(bx, by, bx + 2, by + 2).filter(isShade);
+				if (clist.length < 4) {
+					continue;
+				}
+
+				var isPartOfBiggerLoop = clist.some(function(cell) {
```

Comment:
This check could be simplified to measuring the size of the block the first cell is in.

Reply:
100% agreed, I just had a really hard time determining how to actually access that information and eventually gave up. Just got it though. This made reusing `check2x2Block` actually feasible - it performs a few redundant comparisons but I figure it's worth it for readability's sake.